src/ui/public/visualize/loader/embedded_visualize_handler.js

```diff
@@ -20,6 +20,32 @@ export class EmbeddedVisualizeHandler {
     });
   }
 
+  /**
+   * Update properties of the embedded visualization. This method does not allow
+   * updating all initial parameters, but only a subset of the ones allowed
+   * in {@link VisualizeLoaderParams}.
+   *
+   * @param {Object} [params={}] The parameters that should be updated.
+   * @property {Object} [timeRange] A new time range for this visualization.
+   * @property {Object} [dataAttrs] An object of data attributes to modify. The
+   *    key will be the name of the data attribute and the value the value that
+   *    attribute will get. Use null to remove a specific data attribute from the visualization.
+   */
+  update(params = {}) {
+    this._scope.$evalAsync(() => {
+      if (params.timeRange) {
```

Comment:
also, why is timeRange handled here and all the other parameters are not ? (showSpyPanel,  classNames .... i can see why uiState and appState can't be mutated)

Reply:
Because we currently don't need them :-)

From looking at the past bugs in the loader it showed to be not the best solution to introduce APIs, that we don't use ourselves anywhere, since pretty much all of them were buggy in the beginning :-)

So I would tend to only include what we know we are needing now. If we figure out we (or anyone else) needs to change the CSS class names or other stuff, let's introduce that APIs then.

Reply:
as this is something we expose to the outside (plugin authors) i really think we should think past what we use ourselfs right now, and think about usability of this API outside of our usecases, else ppl won't be able to use this and will still need to hack around which kind of defies its purpose.

but i agree we should introduce tests for this as else we will keep braking this without realizing (ike it happened to our previous APIs, (which worked when introduced but stopped soon after)